packages/elements/src/rating/index.ts

```diff
@@ -111,54 +171,118 @@ export class Rating extends BasicElement {
+      this.tabIndex = 0;
+    }
+    else {
+      this.removeAttribute('role');
+      this.removeAttribute('aria-valuemin');
+      this.removeAttribute('aria-valuenow');
+      this.removeAttribute('aria-valuemax');
+      this.removeAttribute('tabindex');
+    }
+  }
+
+  /**
+   * Handles key input
+   * @param event Key down event object
+   * @returns {void}
+   */
+  protected onKeyDown (event: KeyboardEvent): void {
+    if (event.defaultPrevented || !this.interactive) {
       return;
     }
 
-    const valueIndex = this.maxNumber - index;
-    if (this.valuePrevious !== valueIndex) {
-      this.value = valueIndex.toString();
-      // Dispatch Event when value change
+    // Ignore special keys
+    if (event.shiftKey || event.ctrlKey || event.altKey || event.metaKey) {
+      return;
+    }
+
+    switch (event.key) {
+      case 'Right':
+      case 'Up':
+      case 'ArrowRight':
+      case 'ArrowUp':
+        this.stepUp();
+        break;
+      case 'Left':
+      case 'Down':
+      case 'ArrowLeft':
+      case 'ArrowDown':
+        this.stepDown();
+        break;
+      case 'Home':
+        this.updateValue(this.interactiveMinValue);
+        break;
+      case 'End':
+        this.updateValue(this.max);
+        break;
+      default:
+        return;
+    }
+
+    event.preventDefault();
+  }
+
+  /**
+   * Update value and fired value-changed event
+   * @param value value to updated
+   * @returns {void}
+   */
+  private updateValue (value: number | string): void {
+    value = value.toString();
+    if (this.value !== value) {
+      this.value = value;
       this.notifyPropertyChange('value', this.value);
     }
   }
 
   /**
-   * Invoked whenever the element properties are updated
-   * @param changedProperties changed properties
+   * Increases the value of rating by 1
    * @returns {void}
    */
-  protected updated (changedProperties: PropertyValues): void {
-    super.updated(changedProperties);
-    changedProperties.forEach((oldValue, propName) => {
-      if (propName === 'value') {
-        this.value = this.valueNumber.toString();
-        this.computeRating(this.maxNumber, this.valueNumber);
-      }
-      else if (propName === 'max') {
-        this.max = this.maxNumber.toString();
-        this.computeRating(this.maxNumber, this.valueNumber);
-      }
-    });
+  private stepUp (): void {
+    if (this.valueNumber + 1 > this.maxNumber) {
+      return;
+    }
+    this.updateValue(Math.floor(this.valueNumber) + 1);
   }
 
   /**
-   * A `CSSResultGroup` that will be used
-   * to style the host, slotted children
-   * and the internal template of the element.
-   * @returns CSS template
+   * Decrease the value of rating by 1
+   * @returns {void}
    */
-  static get styles (): CSSResultGroup {
-    return css`
-      :host {
-        display: inline-block;
-      }
-    `;
+  private stepDown (): void {
+    if (this.valueNumber - 1 < this.interactiveMinValue) {
```

Comment:
Also in this case, when value is `1.5` stepDown will not be able to update the value.